numpy/random/tests/test_random.py

```diff
@@ -255,6 +255,18 @@ def test_random_integers_max_int(self):
         desired = np.iinfo('l').max
         np.testing.assert_equal(actual, desired)
 
+    def test_random_integers_deprecated(self):
```

Comment:
Because Deprecation warnings are error for development testing and only raised once from a giving location by default, some care needs to be taken here. There is a `_DeprecationTestCase` in `numpy/core/tests/test_deprecations.py` that can be used as a base class and perhaps should be moved into `numpy.testing`, another, somewhat simpler option is
```
        with warnings.catch_warnings():
            warnings.simplefilter("error", DeprecationWarning)
            assert_raises(DeprecationWarning, ...)
```
@seberg Thoughts about moving the `_DeprecationTestCase`?